### ISSUE TITLE
DB migration error from 0.66.0 to 0.72.2 "SqliteError: no such column: evals.author"

### ISSUE THREAD

**Issue by jay0129** [promptfoo/promptfoo#1223]
Updating self-hosted promptfoo from 0.66.0 to 0.72.2 raises error:
```
Failed to read result with ID undefined from database:
SqliteError: no such column: evals.author
```

It seems like running `npm run db:migrate` inside the container is not possible because migrate.ts is not copied to the container.

**Comment by jay0129** [comment]
If I run the following command inside the container:
```
sqlite3 /root/.promptfoo/promptfoo.db <<EOF
ALTER TABLE evals ADD COLUMN author TEXT;
EOF
```
I can manage to update server to `0.72.1`

However, after updating the server to `0.72.2`, the "Evals" tab is unable to open without any error message

**Comment by typpo** [comment]
We definitely need to create a migrations path within the Docker image. Thanks for flagging this.

Your manually executed migration looks correct.  Is there anything in the JS Console or Network tab when you load Evals?

**Comment by jay0129** [comment]
`api/results/` endpoint  returned the following content:
```
{
    "data": [
        {
            "id": "eval-2024-07-24T02:14:27",
            "label": "Agent - chat (eval-2024-07-24T02:14:27)"
        },
        {
            "id": "eval-2024-07-23T01:47:28",
            "label": "eval-2024-07-23T01:47:28"
        }
    ]
}
```
However, the next HTTP request has a URL path of `api/results/undefined/`.

This issue only occurs on the self-hosted app, whereas the local installation with npm works correctly.

**Comment by typpo** [comment]
Hey @jay0129, thanks for digging into this. I have a fix for the `undefined` route in #1257

**Comment by jay0129** [comment]
Hi @typpo , thanks for fixing this. It works fine for 0.73.5.
